File: ANDROID PROGRAMMING WITH JAVA/PROJECTS/VehicleInsuranceApp/app/src/main/java/com/example/vehicleinsuranceapp/MainActivity.java
package com.example.vehicleinsuranceapp;


import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;

import com.example.vehicleinsuranceapp.fragment.claims.ClaimFormFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Load Claim Form Fragment by default
        if (savedInstanceState == null) {
            loadFragment(new ClaimFormFragment());
        }
    }

    // Method to switch fragments (ClaimFormFragment or ClaimHistoryFragment)
    public void loadFragment(androidx.fragment.app.Fragment fragment) {
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.fragment_container, fragment);
        transaction.addToBackStack(null);
        transaction.commit();
    }
}

File: MINI PROJECTS/GROCERY LIST APP/GroceryShoppingListApp/app/src/main/java/com/example/groceryshoppinglistapp/MainActivity.kt
package com.example.groceryshoppinglistapp



import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.groceryshoppinglistapp.ui.HomeFragment


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            loadFragment(HomeFragment())
        }
    }

    private fun loadFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container, fragment)
            .commit()
    }
}
